ponyo/simulate_expression_data.py

```diff
@@ -555,8 +669,88 @@ def shift_template_experiment(
     # Gene expression data for selected samples
     selected_data_df = normalized_data.loc[sample_ids]
 
+    simulated_data_decoded_df, simulated_data_encoded_df = run_shift_template(loaded_model,
+                                                                              loaded_decode_model,
+                                                                              normalized_data,
+                                                                              selected_data_df,
+                                                                              latent_dim)
+
+    # Un-normalize the data in order to run DE analysis downstream
+    simulated_data_scaled = scaler.inverse_transform(simulated_data_decoded_df)
+
+    simulated_data_scaled_df = pd.DataFrame(
+        simulated_data_scaled,
+        columns=simulated_data_decoded_df.columns,
+        index=simulated_data_decoded_df.index,
+    )
+
+    # Save template data for visualization validation
+    test_filename = os.path.join(
+        local_dir,
+        "pseudo_experiment",
+        "template_normalized_data_" + selected_experiment_id + "_test.txt",
+    )
+
+    selected_data_df.to_csv(test_filename, float_format="%.3f", sep="\t")
+
+    # Save
```

Comment:
Did you mean for this to be a return statement?

Reply:
It was saved in a file in the original code, so I didn't change it https://github.com/greenelab/ponyo/blob/master/ponyo/simulate_expression_data.py

It may be supposed to be a return statement though, the comments seem to imply that it would be

Reply:
The reason I asked is because the other methods return the df as opposed to writing out to files so I was wondering why there was a difference but it looks like I originally wrote it like that not you. To continue to separate out the IO you could change this to return the 2 dfs or I'm happy to update this in another PR.

Reply:
I would lean towards putting it in a different PR. The refactored version gives people an option to return the dfs instead of saving them and I don't want to tinker with the behavior of your original functions in case it breaks something